internal/konnect/node_agent.go

```diff
@@ -1,15 +1,41 @@
 package konnect
 
 import (
+	"context"
 	"fmt"
 	"time"
 
 	"github.com/go-logr/logr"
 
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane"
 	"github.com/kong/kubernetes-ingress-controller/v2/internal/util"
 )
 
-const defaultRefreshNodeInterval = 30 * time.Second
+const (
+	MinRefreshNodePeriod     = 30 * time.Second
+	DefaultRefreshNodePeriod = 60 * time.Second
+	NodeOutdateInterval      = 5 * time.Minute
+)
+
+// ConfigStatusSubscriber subscribes status of configuring kong.
+// REVIEW: define the subscriber here, or internal/adminapi for common usage?
+type ConfigStatusSubscriber interface {
+	Subscribe() chan dataplane.ConfigStatus
+}
+
+type configStatusSubscriber struct {
+	ch chan dataplane.ConfigStatus
+}
+
+var _ ConfigStatusSubscriber = &configStatusSubscriber{}
+
+func (s *configStatusSubscriber) Subscribe() chan dataplane.ConfigStatus {
+	return s.ch
+}
+
+func NewConfigStatusSubscriber(ch chan dataplane.ConfigStatus) ConfigStatusSubscriber {
+	return &configStatusSubscriber{ch: ch}
+}
```

Comment:
I think that the implementation of `Subscribe` and `Notfiy` could be done under one struct as they're using the same channel to provide a communication channel for users of both interfaces. Do you mind moving it to `ChannelConfigNotifier`?

Reply:
done, moved interfaces to `dataplane` and moved `Subscribe` to the same struct.